Maintain session

julianalucena opened the pull request:
Adição de autenticação através da gem Authlogic, a qual foi configurada de modo que a sessão seja compartilhada com a sessão do Redu Core.

julianalucena committed:
Adição da gem Authlogic para autenticação.
- Adição de modelo que representa a Sessão (UserSession).
- Adição de colunas relativas à autenticação ao Usuário.
- Adição de helpers para identificar usuário e sessão atuais.

julianalucena committed:
Adição de controladores e afins para autenticação e criação da sessão do usuário.
- Criação de controlador para Autenticação do usuário (UserSessions).
- Possibilidade de logar tanto por login como por email.
- Adição de helper para mostrar erros no formulário.

julianalucena committed:
Adição das configurações necessárias para manter a sessão entre o Portal de Aplicativos e o Redu Core.
- Utilização do mesmo método de encriptação do Core.
- Utilização do mesmo secret_token do Core.
- Configuração para o cookie ser restrito ao domínio redu.com.br.

julianalucena committed:
Corrige variáveis do js.erb para serem locais.

julianalucena committed:
Merge branch 'united' into maintain-session

julianalucena committed:
Configura Authlogic para utilizar uid na Sessão (desta forma o usuário também é logado no Core).

julianalucena committed:
Adiciona campos necessários à autenticação do usuário ao model_data.yml.

Commit:
Fix testes após adição de sessão de usuário com o Authlogic. Closes #82

Commit:
Merge branch 'master' into maintain-session

Conflicts:
	app/views/shared/_nav_global.html.erb
	config/application.rb

Commit:
Remove password e password confirmation de model_data.yml e adiciona, nas hashes de User das specs do Untied, os parâmetros crypted_password e password_salt (as specs do Untied falham quando precisam validar User).

julianalucena committed:
Remove validação de password e necessidade de password_confirmation. Closes #85

julianalucena committed:
Corrige merge 87baa77a305d4406df04082f032d41c9d842ee3d.

julianalucena committed:
Merge branch 'united' into maintain-session

Conflicts:
	Gemfile

julianalucena committed:
Merge branch 'united' into maintain-session

Conflicts:
	config/application.rb
	config/model_data.yml
	db/schema.rb

guiocavalcanti commented on config/initializers/session_store.rb:
```diff
@@ -1,8 +1,13 @@
+if Rails.env.production?
+  ReduApps::Application.config.session_store :active_record_store,
+    :key => '_redu_session',
+    :domain => '.local.redu'
```
Domínio incorreto.

guiocavalcanti commented on db/migrate/20121111172512_create_user_sessions.rb:
```diff
@@ -0,0 +1,16 @@
+class CreateUserSessions < ActiveRecord::Migration
+  def self.up
```
Aqui poderia usar change.

fltiago commented on app/controllers/application_controller.rb:
```diff
@@ -1,9 +1,16 @@
+
+  private
+  def current_user_session
+    return @current_user_session if defined?(@current_user_session)
```
Fazer `@current_user_session ||= UserSession.find` já não seria suficiente?
Сделать `@current_user_session | | = UserSession.find` уже не достаточно?

fltiago commented on app/controllers/application_controller.rb:
```diff
@@ -1,9 +1,16 @@
 
   def current_user
-    User.find_by_uid(session[:user_uid]) || User.last || FactoryGirl.create(:user)
+    return @current_user if defined?(@current_user)
```
O mesmo comentário que o de cima:
`@current_user  ||= current_user_session && current_user_session.user`

fltiago commented on config/locales/pt-BR.yml:
```diff
@@ -236,4 +236,12 @@ pt-BR:
+  authlogic:
+    error_messages:
+      login_blank: login ou e-mail não pode ficar em branco
+      login_not_found: login ou e-mail e senha inválidos
```
E por quê tem senha aqui? Não deveria ser. `login ou e-mail inválidos`.

julianalucena replied:
Não queria deixar claro onde foi o erro. Isto foi mais por questão de segurança mesmo.

fltiago commented on config/locales/pt-BR.yml:
```diff
@@ -236,4 +236,12 @@ pt-BR:
+    error_messages:
+      login_blank: login ou e-mail não pode ficar em branco
+      login_not_found: login ou e-mail e senha inválidos
+      login_invalid: login ou e-mail e senha inválidos
```
Esse aqui também.

fltiago commented on config/locales/pt-BR.yml:
```diff
@@ -236,4 +236,12 @@ pt-BR:
+      login_not_found: login ou e-mail e senha inválidos
+      login_invalid: login ou e-mail e senha inválidos
+      password_blank: senha não pode ficar em branco
+      password_invalid: login ou e-mail e senha inválidos
```
`Senha inválida` não?

fltiago commented on db/schema.rb:
```diff
@@ -159,6 +159,16 @@
 
   add_index "screen_shots", ["app_id"], :name => "index_screen_shots_on_app_id"
 
+  create_table "sessions", :force => true do |t|
```
Esse arquivo deveria estar no master?

julianalucena replied:
No começo do arquivo, um comentário diz que ele deve ser commitado.

fltiago commented on spec/controllers/application_controller_spec.rb:
```diff
@@ -0,0 +1,59 @@
+      end
+    end
+
+    context 'define helper methods' do
```
Liked a lot.

fltiago commented on spec/controllers/user_sessions_controller_spec.rb:
```diff
@@ -0,0 +1,95 @@
+      end
+
+      it 'logs the user in' do
+        post :create, @login_params.merge!(:locale => 'pt-BR')
```
Por quê não coloca diretamente o `:locale => 'pt-BR'` no `@login_params`.

julianalucena replied:
É que eu só queria deixar coisas relativas ao login lá.

fltiago replied:
Meio bizarro você sempre ter que adicionar o locale.

fltiago commented on spec/factories/user_factory.rb:
```diff
@@ -15,4 +17,4 @@
     end
   end
-end
\ No newline at end of file
```
O_O Ok then.

julianalucena replied:
?

julianalucena committed:
- Corrige domínio do cookie.
- Melhora sintaxe da migration.
- Melhora sintaxe da definição de current_user e current_user_session.

embs commented:
@julianalucena, nos passamos quanto à sintaxe das hashes mas estamos utilizando, no repo do Apps, a nova sintaxe (Ruby 1.9.3) `key: value`. Existe, inclusive, um [issue de melhoria relacionado a isso](https://github.com/redu/apps/issues/25) – que será fechado um dia! \o/

julianalucena commented:
Valeu, @embs. Eu estava ciente disso, mas o costume falou mais alto e eu esqueci completamente. u_u
So sorry.
